src/CallbackStream.php

```diff
@@ -172,10 +172,10 @@ public function getMetadata($key = null)
             return $metadata;
         }
 
-        if (! array_key_exists($key, $metadata)) {
-            return null;
+        if (isset($metadata[$key])) {
```

Comment:
`return $metadata[$key] ?? null;`

Reply:
Good idea but after drop php5 ;(

Reply:
Wha?! This is for an improvement: just drop it already

Reply:
But is it not better to merge it as it to current php5 supported line and then, drop php5 and apply you have suggested?

Reply:
If we can optimize for PHP 5.6 today, we can make it even better for the next _major_ release later. I'm fine with this for now.